lib/codegen/meta/base/settings.py

```diff
@@ -91,6 +91,12 @@
         """,
         default=True)
 
+enable_nan_canonicalization = BoolSetting(
+        """
+        Enable NaN canonicalization
```

Comment:
It'd be good to briefly explain what canonicalization means here, and mention the word "deterministic", to help people who need that find it.